src/EndpointV2/Ruleset/RulesetStandardLibrary.php

```diff
@@ -61,17 +61,22 @@ public function not($value)
      */
     public function getAttr($from, $path)
     {
+        if (preg_match('/^\[(\w+)\]$/', $path, $matches)) {
+            $index = is_numeric($matches[1]) ? (int) $matches[1] : $matches[1];
+
+            return $from[$index] ?? null;
+        }
+
         $parts = explode('.', $path);
         foreach ($parts as $part) {
             $sliceIdx = strpos($part, '[');
             if ($sliceIdx !== false) {
                 if (substr($part, -1) !== ']') {
                     return null;
                 }
-                $slice = intval(substr($part, $sliceIdx + 1, strlen($part) - 1));
-                $from = isset($from[substr($part,0, $sliceIdx)][$slice])
-                    ? $from[substr($part,0, $sliceIdx)][$slice]
-                    : null;
+                $slice = (int) substr($part, $sliceIdx + 1, strlen($part) - 1);
+
+                $from = $from[substr($part, 0, $sliceIdx)][$slice] ?? null;
```

Comment:
Could you please move the logic inside of the from[...] into a variable for making the code more clean. As follow:
```php
$fromIndex = substr($part, 0, $sliceIdx);
$from = $from[$fromIndex][$slice] ?? null;
```